### DISCUSSION TITLE
Combined Rigid Body - Static Mesh Collision

### DISCUSSION THREAD

**Discussion by Hochheilige01** [NVIDIA-Omniverse/PhysX#316] [Q&A]
Hi there! 
I have a few meshes (created with createRigidStatic) and dynamic actor that contains 3 shapes (created with createRigidDynamic). And I wonder how to make them collide with each other. My current behavior - dynamic actor spawns in correct place on the mesh, every frame goes down and down, and when it fully under the static mesh it teleports to origin (point 0, 0, 0) and start falling down from this point. When it reaches some point, it goes back to origin and fall again. It looks weird, but I probably set up something wrong.

I didn't make any special set up for this bodies, and my scene just sets gravity and use default shader. Both actors' shapes contain Simulation flag.

I can provide PVD file if it can help

**Comment by vreutskyy** [comment]
Hi @PavelOvchinnikovGA 
It would help if you could provide a snippet/sample reproducing the issue.
Thanks

**Reply by Hochheilige01** [reply]
Will it be enough if I provide pxd2 file from PVD or ovd generated for Omniverse?

**Reply by vreutskyy** [reply]
I don't know. Let's try.

**Reply by Hochheilige01** [reply]
Here you are
[pvd_scenes.zip](https://github.com/user-attachments/files/17309141/pvd_scenes.zip)

**Reply by Hochheilige01** [reply]
The simulation starts not from the beginning, need to skip frames to see the scene

**Reply by vreutskyy** [reply]
I see some of your static meshes have RestOffset -0.5. That's not a valid value, it can't be negative. Do you check the errors reported by PhysX?

**Reply by vreutskyy** [reply]
Well. The docs say it can be negative. So it seems it's not an error. But I'd suggest to try setting it to a positive value as a negative one doesn't make sense for a triangle mesh. In any case it worth trying to see if it changes anything.

**Reply by Hochheilige01** [reply]
Nah, totally the same with a positive and zero parameter. I got this negative one from snippet related to mesh, hoped it will help :D

**Reply by Hochheilige01** [reply]
I also have error callback enabled and it shows nothing

**Reply by vreutskyy** [reply]
It's hard to say without seeing what's happening in the code. I see contacts are generated, but their normal force is 0. Are you using contact modification? Then I see a bunch of tables that are sinking into the floor. As they sink below a certain level they get teleported to the origin. I don't think they can do it on their own. Someone should set their positions to 0. Some trigger or "kill plane"?

**Reply by Hochheilige01** [reply]
You know, I've thought that nothing was moving my objects, but after your last reply I decided to double check this. And you are right there is some code that for some reason moves objects to origin, so it is really not a PhysX problem
